### PR TITLE
Added time option to run the bot between

### PR DESCRIPTION
**Pull request by nivong**
Hey guys, so I decided to put a bit of time, and something, I felt was missing, I based it off of Yepes tweak, but made it better.
Feel free to set the bot to run at whatever hours you would like!
The likes,follows,comments, and unfollows per day are also based on the amount of the time the bot is running. 
THIS IS SETUP FOR BOT_MODE 0 and BOT_MODE 2
I will update more based on tweaks and stuff I feel I personally want or need. I might take in requests. But feel free to use this, and let me know if there are any issues so I can fix them.
Credits: Szymon Szwinge
https://www.facebook.com/groups/instabot/272211949848217/

### CODE REVIEW

**Comment by virginsince1997** [comment]
Thanks for this, I thinking if you should consider modifying mode 3, to scrape users, like their pictures and follow them, with that we have a better chance of getting a follow back, consider what happens to Private Accounts too

**Comment by LevPasha** [comment]
Guys someone test it?

**Comment by nivong** [comment]
im using it and guys on facebook are :) you can test it :) ?

File: example.py [hunk 1]
```diff
@@ -32,7 +34,20 @@
                                        'travel','art','shop','store','sex','toko','jual','online','murah','jam','kaos','case','baju','fashion',
                                         'corp','tas','butik','grosir','karpet','sosis','salon','skin','care','cloth','tech','rental',
                                         'kamera','beauty','express','kredit','collection','impor','preloved','follow','follower','gain',
-                                        '.id','_id','bags'])
+                                        '.id','_id','bags'],
+               bot_running_hour_start = 0,
+               bot_running_hour_end = 24
```
Comment (LevPasha) [inline example.py]: Why there is 0-24

File: src/instabot.py [hunk 2]
```diff
@@ -103,6 +103,10 @@ class InstaBot:
     # For new_auto_mod
     next_iteration = {"Like": 0, "Follow": 0, "Unfollow": 0, "Comments": 0}
 
+	# Running hours
+    bot_running_hour_start = 14
+    bot_running_hour_end = 23
+
```
Comment (LevPasha) [inline src/instabot.py]: There 14-23

File: src/instabot.py [hunk 3]
```diff
@@ -119,20 +123,28 @@ def __init__(self, login, password,
                  proxy="",
                  user_blacklist={},
                  tag_blacklist=[],
-                 unwanted_username_list=[]):
+                 unwanted_username_list=[],
+                 bot_running_hour_start=0,
+                 bot_running_hour_end=23):
```
Comment (LevPasha) [inline src/instabot.py]: And there 0-23?

File: example.py (hunk 1 above)
Comment (SuperTaranta) [inline example.py]: Well thats where the hours are defined for the bot to read from... so 0 -24, would be 24 hours.

File: src/instabot.py (hunk 2 above)
Comment (SuperTaranta) [inline src/instabot.py]: That was just random hours, as the global variables.

File: src/instabot.py (hunk 3 above)
Comment (SuperTaranta) [inline src/instabot.py]: and then this was just, for if no one added the hours, most of the numbers just stayed from testing ect... Probably would be smart to change them all to 0-24 :P

**Comment by SuperTaranta** [comment]
@LevPasha and as per testing, haven't had anyone complain about issues. Obviously the bot will never get to post 1k likes in an hour, because of the 3 second pause timer. But no one would want that, because thats just not natural.

File: src/instabot.py (hunk 3 above)
Comment (LevPasha) [inline src/instabot.py]: @SuperTaranta Yes... @nivong change all to 0-24

**Comment by nivong** [comment]
updated to match 0 - 23 which is valid (0 = 24 so that doesn't make any sense)

**Pull request merged**